File: src/main/java/com/tpicap/cms/AppConfig.java
package com.tpicap.cms;

import java.io.IOException;
import java.io.InputStream;
import java.util.HashMap;
import java.util.Map;
import java.util.Properties;

import software.amazon.awssdk.regions.Region;

public class AppConfig {

    private Map<String, String> appProperties;

    public AppConfig() {
        try (InputStream input = AppConfig.class.getClassLoader().getResourceAsStream("config.properties")) {

            Properties prop = new Properties();

            if (input == null) {
                System.out.println("Sorry, unable to find config.properties");
                return;
            }

            // load a properties file from class path, inside static method
            prop.load(input);

            Map<String, String> map = new HashMap<>();

            for (String propertyName : prop.stringPropertyNames()) {
                String envValue = System.getProperty(propertyName);
                if (envValue == null || envValue.trim().length() == 0)
                    envValue = System.getenv(propertyName);
                if (envValue == null || envValue.trim().length() == 0)
                    envValue = prop.getProperty(propertyName);
                map.put(propertyName, envValue);
            }

            appProperties = Map.copyOf(map);

        } catch (IOException ex) {
            throw new RuntimeException(ex);
        }

    }

    public String getAwsProfile() {
        return appProperties.get("APP_AWS_PROFILE");
    }

    public String getS3BucketName() {
        return appProperties.get("APP_S3_BUCKET");
    }

    public String getDynamoDBTable() {
        return appProperties.get("APP_DYNAMO_DB_TABLE");
    }

    public Region getAwsRegion() {
        return Region.of(appProperties.get("APP_AWS_REGION"));
    }

}

File: python/lambda/app_config.py
from jproperties import Properties

import os
from pathlib import Path


class AppConfig:
    def __init__(self):
        configs = Properties()
        self.prop = dict()
        config_path = Path(__file__).with_name('config.properties')
        with open(config_path, 'rb') as config_file:
            configs.load(config_file)
        for item in configs.items():    
            key = item[0]
            self.prop[key] = os.environ.get(key, item[1].data)            
        print(self.prop)

    def get_db_table(self):
        return self.prop['APP_DYNAMO_DB_TABLE']

    def get_aws_profile(self):
        return self.prop['APP_AWS_PROFILE']